internal/pages/06_finalize.go

```diff
@@ -145,68 +167,62 @@ func installPassthrough(config *configs.Config) {
 	// A lot of linux systems support modprobe along with their own module system
 	// So copy the modprobe files if we have them
 	modprobeFile := fmt.Sprintf("%s/vfio.conf", config.Path.MODPROBE)
-	if fileio.FileExist(modprobeFile) {
-		// Copy initramfs-tools module to system
-		output = configs.CopyToSystem(modprobeFile, "/etc/modprobe.d/vfio.conf")
-		fmt.Printf("%s\n", output)
+
+	// lets hope by now we've already handled any permissions issues...
+	// TODO: verify that we actually can drop the errors on [fileio.FileExist] call below
+
+	if exists, _ := fileio.FileExist(modprobeFile); exists {
+		// Copy initramfs-tools module to system, note that CopyToSystem will log the command and output
+		// as well as check for errors
+		configs.CopyToSystem(config.IsRoot, modprobeFile, "/etc/modprobe.d/vfio.conf")
```

Comment:
I didn't quite catch that we're doing it like that, I'll look into just copying it with golang if we're root

Reply:
> I didn't quite catch that we're doing it like that, I'll look into just copying it with golang if we're root

well the config files are stored inside `config` which resides in the same directory as the binary.
saw no reason to make it plaster them in /etc since this is a tool you often would just run once and remove :)

Reply:
Should work now! added a test case as well

Reply:
and yeah as a side-note, some automation for testing would be amazing

may be something i can look into in the future

Reply:
> i am happy i can at least test run the program inside a VM to catch this even though i dont use a mutable system anymore 😅

I honestly didn't think this would work properly in qemu for test purposes, if you have too many more failures I can just mark this as WIP again and fire up a VM on my end to run it though to spare you the time

Reply:
> and yeah as a side-note, some automation for testing would be amazing
> 
> may be something i can look into in the future

would be awesome, i am still super green at golang, plus this isnt really the easiest project to make tests for since there is manual verification needed.

I am going to refactor the [ls-iommu](https://github.com/hikariknight/ls-iommu) project at a later date (hoping for end of summer if lucky) since it has a lot of ugly code that was mostly written to just have a functional version with some heavy nested ifs (made it just easier for my brain to visualize the logic as i wrote it), in the end that should give a nice performance bump too when that is done.

But its nice to have some extra eyes here on this project since this is the most front facing of the 2.

Reply:
Yeah I spent some time looking through `ls-iommu` and actually started refactoring it a bit, but decided to reel in my ADHD brain back to this repo. I'm happy to help, I know how much of a daunting task it can be to go back and refactor older projects.